src/mesh/mesh.hpp

```diff
@@ -94,6 +94,7 @@ class Mesh {
 
   // data
   bool modified;
+  const bool is_restart;
```

Comment:
I wanted people to be able to easily do different things upon restart vs first initialization, but without having to register two separate, maybe almost identical functions. There might be a better way than storing restart state in the mesh pointer, but it also seemed natural as something inherited from Athena++.

Reply:
What we've typically been doing is to check for `tm.cycle != 0`, which should also be possible in the new callbacks given that the `SimTime` object is passed.

Reply:
Good point---`is_restart` feels cleaner/less ambiguous to me, but I'm fine to remove this and tell people to use `tm.cycle` if preferred.